src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -989,8 +989,22 @@ bool AuraEffect::CheckEffectProc(AuraApplication* aurApp, ProcEventInfo& eventIn
             // Don't proc extra attacks while already processing extra attack spell
             uint32 triggerSpellId = GetSpellInfo()->Effects[GetEffIndex()].TriggerSpell;
             if (SpellInfo const* triggeredSpellInfo = sSpellMgr->GetSpellInfo(triggerSpellId))
-                if (aurApp->GetTarget()->m_extraAttacks && triggeredSpellInfo->HasEffect(SPELL_EFFECT_ADD_EXTRA_ATTACKS))
-                    return false;
+            {
+                if (triggeredSpellInfo->HasEffect(SPELL_EFFECT_ADD_EXTRA_ATTACKS))
+                {
+                    Unit const* source = eventInfo.GetActor();
+                    uint32 lastExtraAttackSpell = source->GetLastExtraAttackSpell();
+
+                    // Patch 1.12.0(?) extra attack abilities can no longer chain proc themselves
+                    if (lastExtraAttackSpell == triggerSpellId)
+                        return false;
+
+                    // Patch 2.2.0 Sword Specialization (Warrior, Rogue) extra attack can no longer proc additional extra attacks
+                    // 3.3.5 Sword Specialization (Warrior), Hack and Slash (Rogue)
+                    if (lastExtraAttackSpell == 16459 || lastExtraAttackSpell == 66923)
```

Comment:
Not so nice to have hardcoded spell ids. Having enums/constexpr is an improvement, is there any other more generic way ?

Reply:
Actually these spells (including talents themselves) are the only ones with spell icon 1462 (no spell family or flags). Not the best way to check spells but still better than hardcoded values I guess.

Reply:
> Actually these spells (including talents themselves) are the only ones with spell icon 1462 (no spell family or flags). Not the best way to check spells but still better than hardcoded values I guess.

Checking spell icons is a horrible way to identify the spell if you are working on custom content - much better to hardcode the spell ID if it really cannot be done in a generic way, at least then the scope is locked to the entries it was intended for.